support/src/main/java/org/jenkinsci/plugins/workflow/support/pickles/FilePathPickle.java

```diff
@@ -86,6 +86,18 @@ protected FilePath tryResolve() {
         // TODO better to use a synchronized accessor
         @Restricted(NoExternalUse.class)
         public static final Map<VirtualChannel,String> channelNames = new WeakHashMap<VirtualChannel,String>();
+
+        static {
+            // Add channels that may already be known to Jenkins. This can be the case
```

Comment:
@jglick We can do that but not sure I see how it's more robust.

As a matter of interest - what is this listener giving anyway i.e. why doesn't `WorkspaceActionImpl` etc just call something that iterates the list of `Computer`s directly off the `Jenkins` instance? It's all in memory already so I don't see the benefit of creating a secondary list.